cpp/include/cudf_test/iterator_utilities.hpp

```diff
@@ -49,7 +49,7 @@ namespace test {
  * @return auto Validity iterator
  */
 template <typename Iter>
-auto iterator_with_null_at(Iter index_start, Iter index_end)
+[[maybe_unused]] auto iterator_with_null_at(Iter index_start, Iter index_end)
```

Comment:
Hah! Well done. C++17 FTW.

Reply:
Sorry, we might have another problem.

With `static`, the function had internal linkage, and could be defined in the header. Removing `static` makes these functions external.

If you have two TUs include `cudf_test/iterator_utilities.hpp` in the same test directory (e.g. in `tests/copying/`), wouldn't these now have a link error?

Reply:
I think this might need to be:
```c++
[[maybe_unused]] static auto iterator_with_null_at(Iter index_start, Iter index_end)
```

Reply:
Good catch. There will be compilation error. I'm adding `static` back.